src/oas3/accessors.ts

```diff
@@ -56,3 +56,25 @@ export function getSecurities(
       .filter(isNonNullable);
   });
 }
+
+/** Collects scopes for each flow in a way that can be used to uniquely identify a security requirement. */
+export function getScopeKeys(scheme: Omit<SecuritySchemeObject, 'type'>): string[] | undefined {
+  if (!scheme.flows) {
+    return undefined;
+  }
+
+  const scopes: string[] = [];
+
+  function collectScopes(flowType: string, flow?: OAuthFlowObject) {
+    for (const scope of Object.keys(flow?.scopes ?? {})) {
+      scopes.push(`${flowType}::${scope}`);
+    }
+  }
+
+  collectScopes('implicit', scheme.flows.implicit);
+  collectScopes('password', scheme.flows.password);
+  collectScopes('clientCredentials', scheme.flows.clientCredentials);
+  collectScopes('authorizationCode', scheme.flows.authorizationCode);
+
+  return scopes;
```

Comment:
Since the goal is to uniquely identify a security requirement, is the order of these scopes significant?  Should they be in some known order in the result?